src/physics/euler.cpp

```diff
@@ -296,12 +292,11 @@ template <int dim, int nstate, typename real>
 inline real Euler<dim,nstate,real>
 ::compute_entropy_measure ( const real density, const real pressure ) const
 {
-    if (density < 0.0) {
-        this->pcout << "WARNING: Negative density encountered! Returning BIG_NUMBER as entropy measure." << std::endl;
-        return BIG_NUMBER;
-    } else {
-        return pressure*pow(density,-gam);
-    }
+    //Copy such that we don't modify the original density that is passed
+    real density_check = density;
+    const bool density_is_positive = check_positive_quantity<real>(density_check, "density");
+    if (density_is_positive)     return pressure*pow(density,-gam);
+    else                         return BIG_NUMBER;
```

Comment:
Maybe move `BIG_NUMBER` to PhysicsBase as a public const variable since its declared twice now; once at the top of this file and once in the PhysicsBase function

Reply:
Sure, see recent commit.